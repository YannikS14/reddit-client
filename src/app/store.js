import { configureStore } from '@reduxjs/toolkit';
import postsReducer from '../features/posts/postsSlice';
import subredditReducer from '../features/subreddits/subredditsSlice';
import searchFieldReducer from '../features/searchField/searchFieldSlice';

export const store = configureStore({
  reducer: {
    posts: postsReducer,
    subreddits: subredditReducer,
    searchField: searchFieldReducer,
  },
});
