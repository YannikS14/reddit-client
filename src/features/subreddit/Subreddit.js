import React from 'react';
import { updateActiveSubreddit } from '../subreddits/subredditsSlice';
import { useDispatch } from 'react-redux';

export default function Subreddit({
  subreddit = { icon: '', title: '' },
  active,
}) {
  const dispatch = useDispatch();

  return (
    <button
      className={`w-full flex items-center p-4 rounded-md transition-colors hover:text-primary ${
        active ? 'bg-gray-100 dark:bg-gray-600' : ''
      }`}
      onClick={() => dispatch(updateActiveSubreddit(subreddit.title))}
    >
      <img
        src={subreddit.icon}
        alt=""
        className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
      />
      <p className="font-semibold">{subreddit.title}</p>
    </button>
  );
}
