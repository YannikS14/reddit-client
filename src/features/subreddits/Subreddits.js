import React from 'react';
import { useSelector } from 'react-redux';
import { selectSubreddits } from './subredditsSlice';
import Subreddit from '../subreddit/Subreddit';

export default function Subreddits() {
  const { subreddits, activeSubreddit } =
    useSelector(selectSubreddits);

  return (
    <section className="sm:mx-4 sm:w-1/3 sm:ml-0 flex-auto">
      <div className="p-6 bg-white sm:rounded-md shadow-lg">
        <h2 className="text-xl font-bold">Subreddits</h2>
        <div className="mt-4">
          {subreddits.map((subreddit) => {
            return (
              <Subreddit
                subreddit={subreddit}
                key={subreddit.title}
                active={
                  activeSubreddit === subreddit.title ? true : false
                }
              />
            );
          })}
        </div>
      </div>
    </section>
  );
}
