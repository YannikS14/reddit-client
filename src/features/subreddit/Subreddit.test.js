import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import Subreddit from './Subreddit';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Subreddits from '../subreddits/Subreddits';

describe('Subreddit component', () => {
  const testSubreddit = {
    icon: 'http://linktoicon.test',
    title: 'Popular',
  };

  it('renders the subreddit data from props', () => {
    render(
      <Provider store={store}>
        <Subreddit subreddit={testSubreddit} />
      </Provider>,
    );
    expect(screen.getByRole('img')).toHaveAttribute(
      'src',
      'http://linktoicon.test',
    );
    expect(screen.getByText('Popular')).toBeInTheDocument();
  });

  it('renders the clicked subreddit as active', async () => {
    render(
      <Provider store={store}>
        <Subreddits>
          <Subreddit />
        </Subreddits>
      </Provider>,
    );

    expect(screen.getAllByRole('button')[0]).toHaveClass(
      'bg-gray-100 dark:bg-gray-600',
    );
    expect(screen.getAllByRole('button')[1]).not.toHaveClass(
      'bg-gray-100 dark:bg-gray-600',
    );

    userEvent.click(screen.getAllByRole('button')[1]);

    expect(screen.getAllByRole('button')[1]).toHaveClass(
      'bg-gray-100 dark:bg-gray-600',
    );
    expect(screen.getAllByRole('button')[0]).not.toHaveClass(
      'bg-gray-100 dark:bg-gray-600',
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Subreddit />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
