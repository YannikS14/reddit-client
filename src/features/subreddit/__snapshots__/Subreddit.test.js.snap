// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subreddit component matches the snapshot 1`] = `
<button
  className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
  onClick={[Function]}
>
  <img
    alt=""
    className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
    src=""
  />
  <p
    className="font-semibold"
  >
    
  </p>
</button>
`;
